assets/src/components/swingsView.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { ReactElement, useContext, useState } from "react"
+import { useRoutes } from "../contexts/routesContext"
+import { SocketContext } from "../contexts/socketContext"
+import { StateDispatchContext } from "../contexts/stateDispatchContext"
+import { isVehicle } from "../models/vehicle"
+import Loading from "./loading"
+import CloseButton from "./closeButton"
+import { flatten, uniq } from "../helpers/array"
+import { ghostIcon, upRightIcon } from "../helpers/icon"
+import { runIdToLabel } from "../helpers/vehicleLabel"
+import useSwings from "../hooks/useSwings"
+import useVehiclesForRunIds from "../hooks/useVehiclesForRunIds"
+import { ByRunId, VehicleOrGhost } from "../realtime"
+import { ByRouteId, Route, Swing } from "../schedule"
+import { selectVehicle, toggleSwingsView } from "../state"
+import {
+  formattedScheduledTime,
+  now,
+  serviceDaySeconds,
+} from "../util/dateTime"
+
+const SwingsView = (): ReactElement<HTMLElement> => {
+  const [, dispatch] = useContext(StateDispatchContext)
```

Comment:
one of these days we should rename this `StateDispatchContext` and friends, and/or break it into smaller pieces, we have so many states and so many dispatch functions now